result/timelineGenerator.js

```diff
@@ -0,0 +1,158 @@
+
+function dataToArray() {
+    let timelineArray = []
+
+    for (let i = 0; i < data.events.length; i++) {
+        contentArray = [];
+        contentToList(data.events[i].htmlContent)
+        timelineArray.push(contentArray)
+    }
+    return timelineArray
+}
+
+function contentToList(content) {
+    //If there is no content in the row then end with this event.
+    if (content === '') return
+
+    let closeIdx, textInRow, nextRow
+
+    let firstStartIdx = content.indexOf('<')
+    let firstCloseIdx = content.indexOf('>')
+
+    let elem = content.substring(firstStartIdx + 1, firstCloseIdx).split(' ')[0];
+
+    if (elem === 'img') {
+        textInRow = ''
+        nextRow =  content.substring(firstCloseIdx + 1)
+    } else if (elem === 'br') {
+        console.log(1)
+    } else {
+        closeIdx = content.indexOf('</' + elem + '>')
+        textInRow = content.substring(firstCloseIdx + 1, closeIdx)
+        nextRow =  content.substring(closeIdx + ('</' + elem + '>').length)
+    }
+
+    contentArray.push([elem, textInRow, extractParams(content.substring(firstStartIdx + elem.length + 1, firstCloseIdx))])
+
+    contentToList(nextRow)
+}
+
+//Add to dict different parameters.
+function extractParams(str) {
+    let dict = {}
+
+    const regex = /([a-z]*)="(.*?)"/g;
+    let params = (str.match(regex))
+
+    if (params) {
+        for (let param of params) {
+            let paramSplit = param.split("=")
+            dict[paramSplit[0].replaceAll('"', '')] = paramSplit[1].replaceAll('"', '') //There are unnecessary quote marks
+        }
+    }
+    return dict
+}
+
+
+
+//Create container for timeline content
+const scriptContainer = document.querySelector(".timelineContainer");
+const timelineDiv = document.createElement("div");
+timelineDiv.classList.add("timeline")
+
+//Data from dataToArray() is collected here.
+let contentArray = [];
+//Read data from data.js and put it into array.
+let eventArray = dataToArray()
+
+//Generate html
+let left = true
+for (const eventContent of eventArray) {
+    timelineDiv.appendChild(addContainer(eventContent, left))
+    left = !left
+}
+
+//Add generated html to timeline div
+scriptContainer.parentNode.insertBefore(timelineDiv, scriptContainer)
+
+
+
+
+
+function addContainer(content ,left) {
+    const newContainer = document.createElement("div");
+
+    if (left) {
+        newContainer.classList.add("container", "left")
+    } else {
+        newContainer.classList.add("container", "right")
+    }
+
+    const newContent = document.createElement("div");
+    newContent.classList.add("content")
+
+    for (const row of content) {
```

Comment:
wouldn't newContent.insertAdjacentHTML("afterBegin", data.events[i].htmlContent) where i is the index of the corresponding event do exactly the same without the need for all the add... functions (and thus also without the need for dataToArray, contentToList and extractParams functions)?